Website/Default.aspx.cs

```diff
@@ -658,6 +658,11 @@ protected override void OnInit(EventArgs e)
                 {
                     var originalurl = Context.Items["UrlRewrite:OriginalUrl"].ToString();
                     CanonicalLinkUrl = originalurl.Replace(PortalSettings.PortalAlias.HTTPAlias, primaryHttpAlias);
+
+                    if (UrlUtils.IsSecureConnectionOrSslOffload(Request))
+                    {
+                        CanonicalLinkUrl = CanonicalLinkUrl.Replace("http://", "https://");
+                    }

```

Comment:
Wouldn't we want this to be outside of the current `if` statement?

Reply:
Well, since this is the only place where the CanonicalLinkUrl is set, I think it makes sense the way it is.